quinn/src/lib.rs

```diff
@@ -481,137 +414,91 @@ impl EndpointInner {
 
     fn drive_send(&mut self) -> Result<bool, io::Error> {
         let mut blocked = false;
-        if let Some(ref x) = self.outgoing {
-            match self.socket.poll_send(&x.destination, x.ecn, &x.packet) {
-                Ok(Async::Ready(_)) => {
-                    self.outgoing = None;
-                }
+        while let Some(t) = self.outgoing.pop_front() {
```

Comment:
Should guard this loop against more than `IO_LOOP_BOUND` iterations.

Reply:
Done.